symfony/assets/scripts/entry/general.ts

```diff
@@ -32,3 +33,43 @@ jQuery(() => {
     config.save();
   });
 });
+
+jQuery(() => {
+  const newsNavlink = jQuery("#navlink-news");
+
+  // If we're ON the events page, then never show a badge
+  if (newsNavlink.hasClass("active")) {
+    return;
+  }
+
+  // Validate that we have a timestamp for when events were last updated
+  const timestampStr = newsNavlink.data("latest-event");
+  if (!timestampStr) {
+    return;
+  }
+
+  const timestamp = new Date(timestampStr);
+  if (isNaN(timestamp.valueOf())) {
+    return;
+  }
+
+  // If the user has never visited the events page, don't show the badge.
+  // It would be annoying to show up to a new website for the first time and be
+  // told you have unread notifications.
+  const { lastVisitedEventsPage } = Visitor;
+  if (!lastVisitedEventsPage) {
+    return;
+  }
+
+  // If there've been no new events since they last visited, then don't display
+  // the badge
+  if (lastVisitedEventsPage.valueOf() >= timestamp.valueOf()) {
+    return;
+  }
+
+  // There are new events. Show an unread badge
+  const icon = newsNavlink.find("i");
+  const badge = jQuery(document.createElement("div"));
+  badge.addClass("unread-badge");
+  icon.append(badge);
```

Comment:
Why wouldn't you use something like (writing from memory, may be broken code) `icon.append('<div class="unread-badge"/>');`? I suspect that your code could be faster, however this is done once per page load, so simplicity could be better here?

I also wonder, as a person not experienced on frontents, why is it OK to inject a block element into a line element, why wouldn't you use span instead of a div?

Reply:
There were a couple of reasons for my choice here:
1. Performance (you can bypass the string parsing portion)
2. Personal preference for doing one operation at a time where possible
    * SRE principle, but also when something fails/causes an error, you get a more specific location for what went wrong rather than "this string is invalid"
3. Strings don't offer as much type safety/static analysis opportunities
    * TypeScript won't parse/validate the string — so it's easier to have something sneak in
    * A theoretical ESLint plugin could validate that all of your CSS classes must exist in an **.scss** file (look out for typos). This would be pretty trivial to do when you're using functions like `.addClass`, but a lot harder when you're using a composite string (you'd need to teach ESLint how to parse the string as HTML)
4. Easier to refactor/change in the future (at least IMO), since you can change the nodes around without needing to have it all be in a single string

You're absolutely right though — this is happening one time, at page load, and this isn't a complex HTML string. This is just a personal preference, and is also me wanting to ensure I have a consistent style throughout a codebase (vs. a style where I could use `.append("<div />")` for simple things and then `document.createElement` for complex things).

Reply:
As for the block-element-in-inline element thing, that's probably not perfect. It's mostly a habit I've picked up over the years. But changing something from block → inline is so easily done. and I don't think(?) it has any impact on a11y.

In this case, the `<div />` is being absolutely positioned anyway so it's sorta outside the primary document block/inline layout anyways.

tl;dr tho, bad habits die hard